mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp

```diff
@@ -120,6 +120,32 @@ void mlir::sycl::printMemoryTargetMode(AsmPrinter &Printer,
   Printer << memoryTargetModeAsString(MemTargetMode);
 }
 
+std::string
+mlir::sycl::accessAddressSpaceAsString(mlir::sycl::AccessAddrSpace AccAddress) {
+  return std::to_string((int)AccAddress);
+}
+
+mlir::LogicalResult mlir::sycl::parseAccessAddrSpace(
+    mlir::AsmParser &Parser,
+    mlir::FailureOr<mlir::sycl::AccessAddrSpace> &AccAddress) {
+
+  int AddSpaceInt;
+  if (Parser.parseInteger<int>(AddSpaceInt)) {
+    return mlir::ParseResult::failure();
+  }
+
+  assert((0 <= AddSpaceInt <= 6) &&
+         "Expecting address space value between 0 and 6 (inclusive)");
```

Comment:
```suggestion
  assert(0 <= AddSpaceInt && AddSpaceInt <= 6 &&
         "Expecting address space value between 0 and 6 (inclusive)");
```

This was evaluating `(0 <= AddSpaceInt) <= 6`, which is always true as a Boolean value will be casted to either `0` or `1`.